xcube/cli/_gen2/callback.py

```diff
@@ -30,17 +35,48 @@
 
 
 class CallbackApiProgressObserver(ProgressObserver):
-    def __init__(self, callback_api_url: str):
-        self.callback_api_url = callback_api_url
+    def __init__(self, callback_api_cfg: Request):
+        assert_given(callback_api_cfg)
+        assert_given(callback_api_cfg.callback)
+        assert_condition(callback_api_cfg.callback.api_uri and callback_api_cfg.callback.access_token,
+                         "Both, api_uri and access_token must be given.")
+        self.callback_api_cfg = callback_api_cfg
+
+    def _send_request(self, event: str, state_stack: Sequence[ProgressState]):
+        assert_given(state_stack, "ProgressStates")
+        state_stack = state_stack[0].to_dict()
+        callback = {"event": event, "state": state_stack}
```

Comment:
As I said in the Skype chat this morning: The whole progress state stack is low level and may be too much information. You can safely aggregate the stack here to generate a light-weigt, condensed, easy to interprete status info.